### ISSUE TITLE
Band Hopping - Error-21

### ISSUE THREAD

**Issue by makidoja** [dg5lp/JTDX-Helper-159#2]
Hi

When I try to use band hopping i get an error message that Error-21 Band Change Menu is Blocked

![image](https://github.com/dg5lp/JTDX-Helper-159/assets/50527460/c2f9b6cc-d952-461a-92b2-91606c7effaa)

**Comment by dg5lp** [comment]
Hi,

 

It seems that the band-change button is not in the correct position.

Try to move the horizontal tiling as shown below.  Menu must be selected!!!

**Comment by dg5lp** [comment]
Hi,

 

sorry, the vertical tiling was meant…

**Comment by makidoja** [comment]
![image](https://github.com/dg5lp/JTDX-Helper-159/assets/50527460/c727c6f6-fa60-480c-9a6b-7b253be3380e)
![image](https://github.com/dg5lp/JTDX-Helper-159/assets/50527460/85b57689-e71a-4680-a3b3-371d61ba96df)

This is how JTDX-Helper sizes them ??

I didn't see anything attached to your message to try?

**Comment by dg5lp** [comment]
Hi,

 

I see that your window manager is different than mine.

- the corners are round

- the background of the checkboxes is blue.

 

Are you using WINDOWS-11 ?

Is it the default window-manager (desktop-style) ?

 

I know, that are some problems with non-standard window-styles like WIN-7 or “classic”.

 

73

DG5LP

**Comment by makidoja** [comment]
Yes this is a new Windows 11 Pro install with no changes to the design or styleJim M0CKEFrom: dg5lp ***@***.***>Sent: Tuesday, May 30, 2023 7:54:05 pmTo: dg5lp/JTDX-Helper-159 ***@***.***>Cc: Jim M0CKE ***@***.***>; Author ***@***.***>Subject: Re: [dg5lp/JTDX-Helper-159] Band Hopping - Error-21 (Issue #2)
Hi,

 

I see that your window manager is different than mine.

- the corners are round

- the background of the checkboxes is blue.

 

Are you using WINDOWS-11 ?

Is it the default window-manager (desktop-style) ?

 

I know, that are some problems with non-standard window-styles like WIN-7 or “classic”.

 

73

DG5LP

**Comment by makidoja** [comment]
I've managed to dissable the round corners, however I still get the same error

![image](https://github.com/dg5lp/JTDX-Helper-159/assets/50527460/4ca8cc8e-0194-40f9-b431-72be72a59ad8)

**Comment by dg5lp** [comment]
I see, that the automatic window splitting is not correct. I could not find a reliable procedure.

You see, if you shift the splitting, the position of the band select button is shifted too.

 

I will upload a new (test) version with a new expected  position of the button, matching 

better when shifting the window.  I will send a short e-mail if ready to test.

 

73,

DG5LP

 

 

 

Von: Jim M0CKE ***@***.*** 
Gesendet: Mittwoch, 31. Mai 2023 10:04
An: dg5lp/JTDX-Helper-159
Cc: dg5lp; Comment
Betreff: Re: [dg5lp/JTDX-Helper-159] Band Hopping - Error-21 (Issue #2)

 

I've managed to dissable the round corners, however I still get the same error

 <https://user-images.githubusercontent.com/50527460/242202759-4ca8cc8e-0194-40f9-b431-72be72a59ad8.png> Das Bild wurde vom Absender entfernt. image

**Comment by makidoja** [comment]
Brilliant thank youI managed to get it to not throw up the error but it was then selecting the wrong band. Is there a list of frequencies that should be in the list? Or does it use the default list?Jim M0CKESent from Outlook for Android

**Comment by dg5lp** [comment]
Hi,

 

I overlapped my JTDX over yours. The splitting (if it works well) should be as on my version.

The question:  If you shrink the Band-Activity window and then start S/P does it set always to the position as visible on your screenshot?

It would mean that the automatic adjusting of the window size, however working, but not correctly.

 

The band selection window in JTDX should contain max. 10 entries but those you have entered in the Helper`s Band configuration:

160 80 40 30 20 17 15 12 10 6.  The order is important. The Helper cannot read the entries in JTDX. If you want to switch to 40m, the

Helper selects the third entry from the top of the selection list without the possibility to check which entry is there. If it does not 

Match, it can select the bad band, however no exception should happen…

 

DG5LP

 



Von: Jim M0CKE ***@***.*** 
Gesendet: Mittwoch, 31. Mai 2023 12:56
An: dg5lp/JTDX-Helper-159
Cc: dg5lp; Comment
Betreff: Re: [dg5lp/JTDX-Helper-159] Band Hopping - Error-21 (Issue #2)

 

Brilliant thank youI managed to get it to not throw up the error but it was then selecting the wrong band. Is there a list of frequencies that should be in the list? Or does it use the default list?Jim M0CKESent from Outlook for Android

**Comment by makidoja** [comment]
Hi

OK so now it works, however your correct, the windows are resized correctly but if I resize the band activity this stays the same and is not altered

This is where it is working
![image](https://github.com/dg5lp/JTDX-Helper-159/assets/50527460/a324fd0d-adbb-4a77-8d54-4373d47748b0)

If I resize the activity window divider then it stays the same
![image](https://github.com/dg5lp/JTDX-Helper-159/assets/50527460/279ad596-67a4-4fe0-a44d-609d6fddb6f4)

Window has to be so that the band drop down is as far right as possible
![image](https://github.com/dg5lp/JTDX-Helper-159/assets/50527460/b95c8271-6663-4a4b-8111-185ff59c7535)

Also band list needs to be ammended as you said
![image](https://github.com/dg5lp/JTDX-Helper-159/assets/50527460/2a0afe35-a1f4-4c14-ac62-e76d374a2b6f)

Seems to work now, but not from fresh install, from frech install the Band Activity divider has to be moved to the very right as far as it will go.

Jim M0CKE

**Comment by dg5lp** [comment]
Ok, splitting is only resized when starting a S/P or any mode…

 

Von: Jim M0CKE ***@***.*** 
Gesendet: Mittwoch, 31. Mai 2023 14:22
An: dg5lp/JTDX-Helper-159
Cc: dg5lp; Comment
Betreff: Re: [dg5lp/JTDX-Helper-159] Band Hopping - Error-21 (Issue #2)

 

Hi

OK so now it works, however your correct, the windows are resized correctly but if I resize the band activity this stays the same and is not altered

This is where it is working
 <https://user-images.githubusercontent.com/50527460/242272070-a324fd0d-adbb-4a77-8d54-4373d47748b0.png> image

If I resize the activity window divider then it stays the same
 <https://user-images.githubusercontent.com/50527460/242272284-279ad596-67a4-4fe0-a44d-609d6fddb6f4.png> image

Window has to be so that the band drop down is as far right as possible
 <https://user-images.githubusercontent.com/50527460/242272459-b95c8271-6663-4a4b-8111-185ff59c7535.png> image

Also band list needs to be ammended as you said
 <https://user-images.githubusercontent.com/50527460/242272589-2a0afe35-a1f4-4c14-ac62-e76d374a2b6f.png> image

Seems to work now, but not from fresh install, from frech install the Band Activity divider has to be moved to the very right as far as it will go.

Jim M0CKE

**Comment by makidoja** [comment]
Let me check that when I get inSent from Outlook for Android
